### DISCUSSION TITLE
BT-531-Part is allowed in all forms, is it correct?

### DISCUSSION THREAD

**Discussion by eduardsmirnov** [OP-TED/eForms-SDK#257] [Q&A] [answered]
All SDK versions, including the latest one (1.4.1) make the field BT-531-Part visible in all forms (missing forbidden condition).
Is it a bug or expected behavior? It leads to the duplicated field of "Additional nature of the contract". It is also the only field with a suffix "-PART" which is allowed in all forms.

**Comment by YvesJo** [comment, marked as answer]
Indeed, it should only be allowed for PIN Only and forbidden without condition for any other form.
This will get fixed.

**Reply by eduardsmirnov** [reply]
Got the latest SDK v.1.5.0 now. Do I see it correctly that now field BT-531-Part is allowed for ALL forms? It looks bad.

`{
    "id" : "BT-531-Part",
    "parentNodeId" : "ND-PartProcurementScope",
    "name" : "Additional Nature (different from Main)",
    "btId" : "BT-531",
    "xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Part']/cac:ProcurementProject/cac:ProcurementAdditionalType/cbc:ProcurementTypeCode[@listName='contract-nature']",
    "xpathRelative" : "cac:ProcurementAdditionalType/cbc:ProcurementTypeCode[@listName='contract-nature']",
    "type" : "code",
    "legalType" : "CODE",
    "repeatable" : {
      "value" : true,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "codeList" : {
      "value" : {
        "id" : "eforms-contract-nature",
        "type" : "flat",
        "parentId" : "contract-nature"
      },
      "severity" : "ERROR"
    }
  }`

**Reply by YvesJo** [reply]
Indeed, the fix couldn't be applied to 1.5.0 and I expect to have it in the next release. Sorry for the inconvenient.

**Reply by eduardsmirnov** [reply]
Could you check fields.json once again?
Based on current information, both BT-531-Lot and BT-531-Part are visible in forms 4,5,6 due to no limitation for BT-531-Lot

**Reply by YvesJo** [reply]
Thanks for spotting that. Some inactive defined conditional rules makes it currently optional, rules will be fixed with the next release.
